sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -293,8 +299,21 @@ def encode_to_stream(self, value, stream, nested):
     if value is None:
       stream.write_byte(NONE_TYPE)
     elif t is int:
-      stream.write_byte(INT_TYPE)
-      stream.write_var_int64(value)
+      # In Python 3, an int may be larger than 64 bits.
+      # Note that an OverflowError on stream.write_var_int64 would happen
+      # *after* the marker byte is written, so we must check earlier.
+      try:
+        # This may throw an overflow error when compiled.
+        int_value = value
+        # Otherwise, we must check ourselves.
+        if not is_compiled:
+          if not fits_in_64_bits(value):
```

Comment:
Why don't we always check ourselves? Is there a performance hit that visualizes in coders microbenchmarks if we add the check?

Reply:
Yep. Current code:
```
small_int, FastPrimitivesCoder, 1000 element(s): per element median time cost: 1.1301e-07 sec, relative std: 19.00%
```
Removing the is_compiled check
```
small_int, FastPrimitivesCoder, 1000 element(s): per element median time cost: 1.88589e-07 sec, relative std: 18.08%
```

That's over a 60% increase. I tried inlining it and using constants rather than computing the bounds each time, which helps some but the check is entirely redundant on compiled code.

Reply:
Thanks. Consider the following wording for comments:                            
                                                                                
```                                                                             
      # In Python 3, an int may be larger than 64 bits.                         
      # We need to check whether value fits into a 64 bit integer before        
      # writing the marker byte.                                                
      try:                                                                      
        # In Cython-compiled code this will throw an overflow error             
        # when value does not fit into int64.                                   
        int_value = value                                                       
        # If Cython is not used, we must do a (slower) check ourselves.         
        if not is_compiled:                                                     
           ...                                                                                      
```

Reply:
That wording is clearer; thanks.